numba_cuda/numba/cuda/tests/cudadrv/test_nvjitlink.py

```diff
@@ -251,5 +260,64 @@ def kernel():
                 pass
 
 
+@unittest.skipIf(
+    not PYNVJITLINK_INSTALLED, reason="Pynvjitlink is not installed"
+)
+class TestLinkerUsage(CUDATestCase):
+    """Test that whether pynvjitlink can be enabled by both environment variable
+    and modification of config at runtime.
+    """
+    def test_linker_enabled_envvar(self):
+        # Linkable code is only supported via pynvjitlink
+        src = """if 1:
+        import os
+        from numba import cuda
+
+        TEST_BIN_DIR = os.getenv("NUMBA_CUDA_TEST_BIN_DIR")
+        if TEST_BIN_DIR:
+            test_device_functions_cubin = os.path.join(
+                TEST_BIN_DIR, "test_device_functions.cubin"
+            )
+        print(TEST_BIN_DIR)
+        files = (
+            test_device_functions_cubin,
+        )
+        for lto in [True, False]:
+            for file in files:
+                sig = "uint32(uint32, uint32)"
+                add_from_numba = cuda.declare_device("add_from_numba", sig)
+
+                @cuda.jit(link=[file], lto=lto)
+                def kernel(result):
+                    result[0] = add_from_numba(1, 2)
+
+                result = cuda.device_array(1)
+                kernel[1, 1](result)
+                assert result[0] == 3
+        """
+        env = os.environ.copy()
+        env['NUMBA_CUDA_ENABLE_PYNVJITLINK'] = "1"
+        print(env['NUMBA_CUDA_TEST_BIN_DIR'])
+        run_in_subprocess(src, env=env)
+
+    def test_linker_enabled_config(self):
```

Comment:
I think this test also needs to run in a subprocess where the `NUMBA_CUDA_ENABLE_PYNVJITLINK` env var is not present. The CI setup at present will already be running this test with that environment variable set to `1` in the cases where this test actually runs - see e.g.

https://github.com/NVIDIA/numba-cuda/blob/f57706fdce83f51f7dd65ad5ed435810299605b2/ci/test_conda_pynvjitlink.sh#L72

Reply:
It may also be good to have a test of where the config variable is used to disable pynvjitlink, with `CUDA_ENABLE_PYNVJITLINK` set to `False` to ensure it's successfully disabled.

Reply:
Isn't that already exist? 

https://github.com/isVoid/numba-cuda/blob/c9d148b781b9a2d941d4e64e1b8215cbc6d333a2/ci/test_conda.sh#L51-L52